src/main/java/junitparams/internal/parameters/ParametersFromCustomProvider.java

```diff
@@ -28,11 +31,28 @@ public boolean isApplicable() {
     }
 
     private ParametersProvider instantiate(Class<? extends ParametersProvider> providerClass) {
-        try {
-            return providerClass.newInstance();
-        } catch (Exception e) {
-            throw new RuntimeException("Your Provider class must have a public no-arg constructor!", e);
+        Constructor<?>[] constructors = providerClass.getConstructors();
+        for (Constructor<?> constructor : constructors) {
```

Comment:
```java
class MyType {
  public MyType(FrameworkMethod frameworkMethod, String token) {
    this.frameworkMethod = frameworkMethod;
    this.token = token;
  }

  public MyType(FrameworkMethod frameworkMethod) {
    this(frameworkMethod, "SECRET")
  }

  ...
}
```

there, a secondary constructor that would fail but shouldn't

likewise the following will fail also

```java
class MyType {
  public MyType(String token) {
    this.token = token;
  }

  public MyType(FrameworkMethod frameworkMethod) {
    this("SECRET")
  }

  ...
}
```

code should be tolerant of all possible conditions failing only where it *is* exceptional